src/deluge/gui/views/arranger_view.cpp

```diff
@@ -1273,17 +1273,21 @@ void ArrangerView::editPadAction(int32_t x, int32_t y, bool on) {
 
 			// Already pressing - length edit
 			else if (currentUIMode == UI_MODE_HOLDING_ARRANGEMENT_ROW) {
-
+				// press on different row than currently holding pad
 				if (y != yPressedEffective) {
 					if (!pressedClipInstanceIsInValidPosition) {
 						return;
 					}
+					// create new clip instance on new press
 					return createNewClipInstance(output, x, y, squareStart, squareEnd, xScroll);
 				}
-
+				// press on same row, to the right of currently holding pad
 				if (x > xPressed) {
-					adjustClipInstanceLength(output, xPressed, y, squareStart, squareEnd);
+					// change length of selected clip instance
+					return adjustClipInstanceLength(output, xPressed, y, squareStart, squareEnd);
 				}
+				// press on same row, to the left of the currently holding pad, do nothing
+				return;
```

Comment:
Return behavior implementation is a bit inconsistent - this is not absolutely needed here but doesn't hurt either